src/xunit.v3.templates/templates/project/xunit3-extension/fs/SupportedOSAttribute.fs

```diff
@@ -37,16 +37,19 @@ type SupportedOSAttribute([<ParamArray>]supportedOSes: SupportedOS[]) =
 
         for supportedOS in supportedOSes do
             let osPlatform =
-                match osMappings.TryGetValue supportedOS with
-                | true, value -> value
-                | _ -> raise (new ArgumentException($"Supported OS value '{supportedOS}' is not a known OS", nameof(supportedOSes)))
+                match Map.tryFind supportedOS osMappings with
+                | Some value -> value
+                | None -> invalidArg (nameof supportedOSes) $"Supported OS value '{supportedOS}' is not a known OS"
```

Comment:
for argumentexception, the `name` should refer to the argument with the bad value, i.e.

| None -> invalidArg (nameof supportedOS) $"Supported OS value '{supportedOS}' is not a known OS"